src/msg/msg_types.h

```diff
@@ -245,6 +249,8 @@ struct entity_addr_t {
   bool is_legacy() const { return type == TYPE_LEGACY; }
   bool is_msgr2() const { return type == TYPE_MSGR2; }
   bool is_any() const { return type == TYPE_ANY; }
+  // this isn't a guarantee; some client addrs will match it
+  bool maybe_cidr() const { return get_port() == 0 && nonce != 0; }
```

Comment:
There are other solutions here that won't be ambiguous:

For IPv4: check that the first octet is `224` (multicast). If so, use the port as that octet and nonce as CIDR block range.
For IPv6: check that the first octet is `ff00` (also multicast). As with IPv4, use the port as that octet and nonce as CIDR block range.

Reply:
Those suggestions are way more intrusive to the logic of entity_addr_t — at that point I'd probably create a new type.

In the final logic this is just used by OSDMonitor to make sure the parsed address makes sense as a CIDR address, at which point we set TYPE_CIDR and it's unambiguous.

Reply:
Got it!